src/main/java/data/Database.java

```diff
@@ -55,25 +62,17 @@ public OutputStream writeData(String runId, String type) {
     return null;
   }
 
-  private String makeFileName(String runId, String type) {
-    return getDatabase() + "/" + runId + "_" + type;
-  }
-
-  /* Adds file to database given its runid and type. */
-  public String newDatabaseEntry(String runId, String type) {
-    String fileName = makeFileName(runId, type);
-    files.add(fileName);
-    return fileName;
-  }
-
-  /* Return name of file with runid and type in database. */
-  public String findName(String runId, String type) {
-    String fileName = getDatabase() + "/" + runId + "_" + type;
-    return files.get(0);
+  /*Returns appropiate file name for data with this runId and type. */
+  private String fileName(String runId, String type) {
+    return getDatabaseName() + "/" + runId + "_" + type;
   }
 
   /* Returns list of files in database. */
   public ArrayList<String> getAllFiles() {
-    return files;
+    File[] files = getDatabase().listFiles();
+    List<File> filesAsFiles = Arrays.asList(files);
```

Comment:
I think you don't need to create another list, you can just use files
for (int i = 0; i < files.length; i++) {
   filesAsStrings.add(files[i].getName());
}